sql/parser/builtins.go

```diff
@@ -851,24 +851,24 @@ var substringImpls = []builtin{
 		types:      typeList{stringType, intType},
 		returnType: DummyString,
 		fn: func(_ EvalContext, args DTuple) (Datum, error) {
-			str := args[0].(DString)
+			runes := []rune(string(args[0].(DString)))
```

Comment:
Do you need the `string()` here? I would think you could go directly from `DString` to `[]rune`. 

Converting the whole string to `[]rune` is inefficient; iterating with `utf8.DecodeRune` would be more efficient. But we probably don't care too much about this now.